fontbe/src/fvar.rs

```diff
@@ -0,0 +1,72 @@
+//! Generates a [fvar](https://learn.microsoft.com/en-us/typography/opentype/spec/fvar) table.
+
+use std::collections::HashMap;
+
+use fontdrasil::orchestration::Work;
+use log::trace;
+use read_fonts::types::{MajorMinor, Tag};
+use write_fonts::tables::fvar::{AxisInstanceArrays, Fvar, VariationAxisRecord};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Context},
+};
+
+const HIDDEN_AXIS: u16 = 0x0001;
+
+struct FvarWork {}
+
+pub fn create_fvar_work() -> Box<BeWork> {
+    Box::new(FvarWork {})
+}
+
+impl Work<Context, Error> for FvarWork {
+    /// Generate [fvar](https://learn.microsoft.com/en-us/typography/opentype/spec/fvar)
+    fn exec(&self, context: &Context) -> Result<(), Error> {
+        let static_metadata = context.ir.get_init_static_metadata();
+        // Guard clause: don't produce fvar for a static font
+        if static_metadata.axes.is_empty() {
```

Comment:
note that static_metadata.axes currently is currently never empty when building from a designspace, so we always emit an fvar even when the input contain no variations.
A designspace without any axes makes no sense of course, there's always going to be at least one.
It's not merely the presence of an axis that makes a font variable, but whether those axes contain any variations. E.g. take our static.designspace test file, it has one axis, one source (master) and one instance.. it's effectively a single UFO with a no-op accompanying designspace file whose only purpose is acting as an input for fontc (which currently only supports either .glyphs or .designspace inputs, no single .ufo).
When running fontc on such a static.designspace file, I would expect that no fvar be built, even if a Weight axis is declared in the designspace. That's because there are actually no variations for Weight in this designspace. You can detect this situation by either checking that an axis min==default==max (as in this particular case), or also by seeing if there are more than 1 source (or master) alongside the given axis; if there's only one, there can't be any variation.
I'm not sure about .glyphs file whether it defaults to no axes or implies at least weight, will need to test.

Reply:
Quite right, I need to filter out point axes. Will send a followon.